prometheus_es_exporter/__init__.py

```diff
@@ -509,6 +515,18 @@ def cli(**options):
                                    '--indices-stats-mode must be "indices" for '
                                    '--indices-stats-indices to be used.')
 
+    if options['header']:
+        def parse_header(header):
+            """Split header into name and value"""
+            parts = tuple(part.strip() for part in header.split(":", 1))
+            if len(parts) != 2 or any(part == '' for part in parts):
+                raise click.BadOptionUsage('header',
+                                           'Invalid header "%s". Use colon to separate name and value.' % header)
+            return parts
+        headers = dict(parse_header(h) for h in options['header'])
```

Comment:
Not essential, but this parsing/validation could be done in a [callback on the click option](https://click.palletsprojects.com/en/7.x/options/#callbacks-for-validation).

There's an [example in the similar prometheus-mysql-exporter project](https://github.com/braedon/prometheus-mysql-exporter/blob/29b2483b4f0153452e9e433ac395d333584f21b7/prometheus_mysql_exporter/__init__.py#L106-L116) if you want to make the switch.

Reply:
Makes sense, I changed that in 22cd35f . Incidentally, it also solved the merge conflict.